### PR TITLE
Added .gitignore, MathPlugin.vcxproj, sdk. Updated sampgdk.

### PR DESCRIPTION
**Pull request by ikkentim**
Saw in a sampgdk issue you needed your sampgdk updated in this plugin.

There are a few things to note about this PR:
- I've only tested it on Windows
- `makefile` has not been updated
- I use a newer version of visual studio, therefore I had to update the solution to a `.vcxproj` file.
- I added a `.gitignore` file, which is useful.
- In the version on `master`, you seem to have your sdk/gdk in a directory outside the projects path. Because i don't think that's a logical thing to do and because it's just easier this way, I added it to he project.

Feel free to cherry pick or dismiss this pull request.

### CODE REVIEW

**Comment by JernejL** [comment]
merged. will test linux compile soon.

**Comment by JernejL** [comment]
I can't figure out how and where you intended sampgdk to be placed, can you now explain it & maybe just add sampgdk as external so this can compile out of the box?

**Comment by ikkentim** [comment]
... My bad. sampgdk.h and sampgdk.c were supposed to be in the root directory. I obviously failed to commit those :s I'll commit them once I'm at my pc. Or, you could add them yourself. They are available here https://github.com/Zeex/sampgdk/releases/download/v4.2.2/sampgdk-4.2.2-amalgamation.zip

**Comment by Zeex** [comment]
@ikkentim They are already in the root directory, you didn't forget them.

**Comment by ikkentim** [comment]
Lol, I should get my eyes checked.

**Comment by Zeex** [comment]
@JernejL Are you building on Linux? There's an updated makefile in the latest commit, it should build without additional files.

**Comment by JernejL** [comment]
No, i'm trying to build it on windows with msvc 2008 and am getting classical missing files issues. 

1>c1xx : fatal error C1083: Cannot open source file: '.\samp\plugin\amxplugin.cpp': No such file or directory
1>callbacks.cpp
1>c1xx : fatal error C1083: Cannot open source file: '.\samp\callbacks.cpp': No such file or directory
1>eventhandler.cpp
1>c1xx : fatal error C1083: Cannot open source file: '.\samp\eventhandler.cpp': No such file or directory
1>fakeamx.cpp
1>c1xx : fatal error C1083: Cannot open source file: '.\samp\fakeamx.cpp': No such file or directory

---

1>jump.cpp
1>c1xx : fatal error C1083: Cannot open source file: '.\samp\jump.cpp': No such file or directory
1>logprintf.cpp
1>c1xx : fatal error C1083: Cannot open source file: '.\samp\logprintf.cpp': No such file or directory
1>objects.cpp
1>c1xx : fatal error C1083: Cannot open source file: '.\samp\objects.cpp': No such file or directory
1>players.cpp
1>c1xx : fatal error C1083: Cannot open source file: '.\samp\players.cpp': No such file or directory
1>samp.cpp
1>c1xx : fatal error C1083: Cannot open source file: '.\samp\samp.cpp': No such file or directory
1>vehicles.cpp
1>c1xx : fatal error C1083: Cannot open source file: '.\samp\vehicles.cpp': No such file or directory
1>wrapper.cpp
1>c1xx : fatal error C1083: Cannot open source file: '.\samp\wrapper.cpp': No such file or directory

So basically what do i need to do - where do i place sampgdk now.. what do i need to define in project to get this to compile? 

Could sampgdk be added to git as a submodule or something that this can be just pulled and compile without additional issues for people?

**Comment by ikkentim** [comment]
You are using the wrong project file (MathPlugin.vcproj). As I noted in the PR, I use a newer version of visual studio which created MathPlugin.vcxproj. Either you should try updating your visual studio or fix the vcproj file.

If you decide to update your visual studio, you better delete the vcproj file.

If not, here are the project settings I used to compile the plugin:
http://www.quick-screen.co.uk/juad3r
http://www.quick-screen.co.uk/ds097c
http://www.quick-screen.co.uk/bk89d9
http://www.quick-screen.co.uk/ffbq7o

**Comment by JernejL** [comment]
That would be helpful.. i only got vs2008. what vs did you use?

**Comment by ikkentim** [comment]
vs2013 (profesional, but express edition would do as well)

**Comment by ikkentim** [comment]
speaking of not up to date project files, MathPlugin.cbp isn't up to date either.

**Comment by Zeex** [comment]
They give professional edition for free now (it is called "community" edition), there's really no reason to keep that vcproj file, well unless you're running Windows XP.
